lib/discourse_plugin_registry.rb

```diff
@@ -36,13 +38,14 @@ def self.define_register(register_name, type)
   #   - Defines instance method as a shortcut to the singleton method
   #   - Automatically deletes the register on registry.reset!
   def self.define_filtered_register(register_name)
+    return if respond_to?(register_name)
     define_register(register_name, Array)
 
     singleton_class.alias_method :"_raw_#{register_name}", :"#{register_name}"
 
     define_singleton_method(register_name) do
       unfiltered = public_send(:"_raw_#{register_name}")
-      unfiltered.filter { |v| v[:plugin].enabled? }.map { |v| v[:value] }.uniq
+      unfiltered.filter_map { |h| h[:value] if h[:plugin].enabled? }.uniq
     end
```

Comment:
Different style? Not sure if easier to read / understand 🤔 

```ruby
    define_singleton_method(register_name) do
      public_send(:"_raw_#{register_name}")
        .filter_map { |h| h[:value] if h[:plugin].enabled? }
        .uniq
    end
```

Reply:
You’re right, we don’t need a variable here :+1: